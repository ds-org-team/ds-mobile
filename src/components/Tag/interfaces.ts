export type TagProps = {
  label: string;
  color?: Custom.Colors;
};
