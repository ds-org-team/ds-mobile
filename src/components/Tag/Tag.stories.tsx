import React from 'react';
import { COLORS } from '../../themes/tokens';
import { TagProps } from './interfaces';
import Tag from './Tag';

const [black, white, ...colors] = COLORS;

export default {
  title: 'Componente/Tag',
  component: Tag,
  argTypes: {
    label: {
      control: { type: 'text' },
      defaultValue: 'Label',
    },
    color: {
      control: { type: 'select', options: colors },
    },
    icon: {
      control: { type: 'text' },
      defaultValue: 'account',
    },
    size: {
      control: { type: 'select', options: ['small', 'medium'] },
      defaultValue: 'medium',
    },
  },
};

export const Default = (props: TagProps): React.ReactNode => <Tag {...props} />;
