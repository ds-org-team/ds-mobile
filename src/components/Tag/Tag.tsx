import React from 'react';
import { createRestyleComponent } from '@shopify/restyle';
import { TagProps } from './interfaces';
import { Theme } from '../../themes';
import Text from '../Text';
import Box from '../Box';
import Icon from '../Icon';

const Tag: React.FC<TagProps> = ({ label, size, icon, color }) => {
  const contrastColor = color?.includes('dark') ? 'white' : 'black';
  const style = size === 'medium' ? { height: '26px' } : { height: '20px' };
  const spacins = icon ? { pl: 'quark' } : {};

  return (
    <Box
      py="nano"
      px="nano"
      borderRadius="nano"
      backgroundColor={color || 'neutral-light'}
      flexDirection="row"
      style={{ ...style, width: 'min-content' }}
      display="flex"
      alignItems="center"
      {...spacins}
    >
      {icon && (
        <Icon sz="xs" color={contrastColor} name={icon} op="opacity-4" />
      )}
      <Text
        fs={size === 'small' ? 'sm' : 'md'}
        color={contrastColor}
        op="opacity-4"
      >
        {label}
      </Text>
    </Box>
  );
};
export default createRestyleComponent<TagProps, Theme>([], Tag);
