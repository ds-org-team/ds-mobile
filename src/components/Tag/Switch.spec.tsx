/* eslint-disable global-require */
import React from 'react';
import { cleanup, render } from '@testing-library/react-native';
import { ThemeProvider } from '@shopify/restyle';
import Tag from './Tag';
import themeMaestro from '../../themes/maestro';

afterEach(cleanup);

jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon');

test('should Checkbox render correctly', async () => {
  const { toJSON } = render(
    <ThemeProvider theme={themeMaestro}>
      <Tag label="Label tag" size="small" />
    </ThemeProvider>,
  );

  expect(toJSON()).toMatchSnapshot();
});
