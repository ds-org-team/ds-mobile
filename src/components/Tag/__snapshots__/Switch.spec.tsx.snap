// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Checkbox render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#EEE6F8",
        "borderRadius": 8,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      },
      Array [
        Object {},
      ],
    ]
  }
  testID="Box"
>
  <Text
    style={
      Array [
        Object {
          "color": "#420099",
          "display": "flex",
          "fontFamily": "\\"Inter\\", sans-serif",
          "fontWeight": "500",
          "margin": 4,
        },
        Array [
          Object {
            "fontSize": 10,
          },
        ],
      ]
    }
  >
    Label tag
  </Text>
</View>
`;
